spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Predictor.scala

```diff
@@ -53,21 +55,67 @@ object Predictor {
   (localModel: Module[T], samples: Seq[Sample[T]],
     localToBatch: Transformer[Sample[T], MiniBatch[T]],
     shareBuffer: Boolean,
-    outputLayer: String = null)(implicit ev: TensorNumeric[T]): Iterator[Tensor[T]] = {
+    outputLayer: String = null)(implicit ev: TensorNumeric[T]): Iterator[Activity] = {
     localToBatch(samples.toIterator).flatMap(batch => {
       localModel.forward(batch.getInput())
       val output = if (outputLayer == null) {
-        localModel.output.toTensor[T]
+        localModel.output
       } else {
-        localModel(outputLayer).get.output.toTensor[T]
+        localModel(outputLayer).get.output
```

Comment:
localModel(outputLayer) maybe None.
Please find the outputModel before flatMap.